vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs

```diff
@@ -81,22 +81,24 @@ module private FSharpProjectOptionsHelpers =
     let hasProjectVersionChanged (oldProject: Project) (newProject: Project) =
         oldProject.Version <> newProject.Version
 
-    let hasDependentVersionChanged (oldProject: Project) (newProject: Project) cancellationToken =
-        async {
-            let! oldVersion = oldProject.GetDependentVersionAsync(cancellationToken) |> Async.AwaitTask
-            let! newVersion = newProject.GetDependentVersionAsync(cancellationToken) |> Async.AwaitTask
-            return oldVersion <> newVersion
-        }
+    let hasDependentVersionChanged (oldProject: Project) (newProject: Project) =
+        let oldProjectRefs = oldProject.ProjectReferences
```

Comment:
Do we guarantee that this is stable?

Reply:
With manual testing, yes. Adding a file to a project will correctly invalidate projects that depend on it. And modifying a document doesn't invalidate the document.